garden_ai/notebook_metadata.py

```diff
@@ -221,6 +177,126 @@ def save_requirements_data(
         return None
 
 
+def display_metadata_widget():
+    """
+    Displays the metadata editor widget
+    When one of the widgets fields is changed, pickles and saves the updated NotebookMetadata.
+    When the notebook is saved, the post_save_hook in custom_jupyter_config will
+    go and look for the pickled NotebookMetadata and save it to the notebooks metadata.
+    """
+
+    # NOTEBOOK_PATH env var set in start_container_with_notebook
+    notebook_path = os.environ["NOTEBOOK_PATH"]
+    nb_meta = get_notebook_metadata(notebook_path)
+
+    output = widgets.Output()
+
+    # Global DOI widget
+    doi_widget = widgets.Textarea(
+        value=nb_meta.global_notebook_doi,
+        placeholder="Global DOI",
+        continuous_update=False,
+        disabled=False,
+    )
+
+    def doi_observer(change):
+        with output:
+            nonlocal nb_meta
+            nb_meta.global_notebook_doi = change.new.strip()
+            if nb_meta.global_notebook_doi == "":
+                nb_meta.global_notebook_doi = None
+            _pickle_metadata(nb_meta)
+
+    doi_widget.observe(doi_observer, "value")
+
+    # Base image name widget
+    base_image_widget = widgets.Textarea(
+        value=nb_meta.notebook_image_name,
+        placeholder="Base image name",
+        continuous_update=False,
+        disabled=False,
+    )
+
+    def base_image_observer(change):
+        with output:
+            nonlocal nb_meta
+            nb_meta.notebook_image_name = change.new.strip()
+            if nb_meta.notebook_image_name == "":
+                nb_meta.notebook_image_name = None
+            _pickle_metadata(nb_meta)
+
+    base_image_widget.observe(base_image_observer, "value")
+
+    # Requirements widget
+    req_widget = None
+    if nb_meta.notebook_requirements.file_format == "pip":
+        reqs_string = "\n".join([req for req in nb_meta.notebook_requirements.contents])
+        req_widget = widgets.Textarea(
+            value=reqs_string,
+            placeholder="Requirements",
+            layout=widgets.Layout(width="100%", height="80px"),
+            continuous_update=False,
+            disabled=False,
+        )
+    elif nb_meta.notebook_requirements.file_format == "conda":
+        req_widget = widgets.Textarea(
+            value=yaml.safe_dump(nb_meta.notebook_requirements.contents),
+            placeholder="Requirements",
+            layout=widgets.Layout(width="100%", height="300px"),
+            continuous_update=False,
+            disabled=False,
+        )
```

Comment:
Nothing wrong with this, but a question for @OwenPriceSkelly: are we going to stop supporting conda? If so, we could probably leave this out.